src/sentry/integrations/jira/integration.py

```diff
@@ -204,6 +204,9 @@ def update_organization_config(self, data):
         config.update(data)
         self.org_integration.update(config=config)
 
+    def merge_issue_config_defaults(self, project_id, config):
```

Comment:
Added so that `merge_issue_config_defaults` does not break jira's dynamic fields.

Reply:
Hmm, do we actually have to do this. Would you actually be able to walk me through why this is needed, in this PR in particular too?

Reply:
@EvanPurkhiser, I agree. Originally it wasn't in this pr. It was in a separate PR here https://github.com/getsentry/sentry/pull/9857,  but @macqueen was concerned that having it in two separate prs. I wasn't entirely clear as to why she wanted it this way, but it was something she specifically requested.

Reply:
Looked at this more closely and I actually don't think omitting this breaks Jira in this pr since `get_persisted_default_config_fields` hasn't been defined. I'm still pretty sure that you'd need to override it in the other PR because we get the project id from params sometimes here: https://github.com/getsentry/sentry/pull/9838/files#diff-8515fbe78ce7facb9d8c6069bc5b129bR135 after a user has just changed their project and we're re-fetching the issue metadata, which we'd just overwrite with the merge call, right? or am i missing something.

Reply:
Ah @macqueen thanks for reminding me. I didn't think we followed those rules particularly often. For example, it talks a lot about how pr's should be formatted and most pr's I've seen don't even contain a body. That said this is probably a bit more important. 

I also totally forgot that `get_persisted_default_config_fields` hasn't been defined for jira, so this is not needed. As to your last question, I'll look into it more and add a test in the jira pr to show that it handles that case. I think that's probably the best approach there.